src/linter.ts

```diff
@@ -77,7 +73,8 @@ async function flintCustom(
         dir as keyof Config
       ] as FlinterCustomRuleSettings;
 
-      const currentRule = ruleSetting.frontmatter[index];
+      const currentRule = ruleSetting.frontmatter.find(f => f.field == field) ?? ruleSetting.frontmatter[0];
```

Comment:
Not sure if there is a better way to do this.
`find()` can return undefined but `currentRule` can't be undefined